ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -117,11 +117,10 @@ protected void handleSuccess() {
     final long lastErrorTime = lastError.getAndUpdate(x -> NO_ERROR_TIME);
     if (lastErrorTime == STARTUP_LAST_ERROR_TIME) {
       eventLog.executionClientIsOnline();
-      executionClientEventsPublisher.onAvailabilityUpdated(true);
     } else if (lastErrorTime != NO_ERROR_TIME) {
       eventLog.executionClientRecovered();
-      executionClientEventsPublisher.onAvailabilityUpdated(true);
     }
+    executionClientEventsPublisher.onAvailabilityUpdated(true);
```

Comment:
moving the call here you'll call it on every successful call (which cause some UT to fail). I'd revert it to what was before, unless you have an argument for doing so :)

Reply:
I think I thought about simplifying the code and forgot I planned to avoid unnecessarily publishing the event when we are already available. :)